aequilibrae/paths/results/path_results.py

```diff
@@ -91,13 +104,20 @@ def prepare(self, graph: Graph) -> None:
         self.nodes = graph.num_nodes + 1
         self.zones = graph.centroids + 1
         self.links = graph.num_links + 1
-        self.num_skims = graph.skims.shape[1]
+        self.num_skims = len(graph.skim_fields)
 
         self.predecessors = np.zeros(self.nodes, dtype=self.__integer_type)
         self.connectors = np.zeros(self.nodes, dtype=self.__integer_type)
         self.reached_first = np.zeros(self.nodes, dtype=self.__integer_type)
-        self.skims = np.zeros((self.nodes, self.num_skims), self.__float_type)
+        if self.num_skims:
+            self.skims = np.empty((graph.all_nodes[-1] + 1, self.num_skims), self.__float_type)
+            self.skims.fill(np.inf)
+            self._skimming_array = np.zeros((self.nodes, self.num_skims), self.__float_type)
+        else:
+            self._skimming_array = np.zeros((1, 2), self.__float_type)
+
         self.__graph_id__ = graph.__id__
+        self.__graph_sum = 2 * graph.cost.sum()
```

Comment:
why 2*?

Reply:
Because I can imagine somebody creating a worst-case scenario network (imagining that turn penalties are considered) where one needs to traverse all links (or almost all) in both directions.

Reply:
maybe worth commenting that?